components/service/glean/src/main/java/mozilla/components/service/glean/firstrun/FileFirstRunDetector.kt

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.firstrun
+
+import android.support.annotation.VisibleForTesting
+import mozilla.components.support.base.log.logger.Logger
+import java.io.File
+import java.io.IOException
+
+/**
+ * A [FirstRunDetector] implementation that uses the presence
+ * of a specific file in the User's app directory to determine
+ * if this is a first run or not.
```

Comment:
Nice! This is so generic that I wonder whether this could be moved to `support-utils` (in a later PR). Every component that needs something like this could intiialize this with a dir/file. I assume we do not want to share this "firstrun" state because it's very component-specific though, right?

Reply:
Yeah. I was thinking about the opposite: If we provide a FirstRunThingy for everyone in support-utils or support-base... would this be good enough or does "first run" mean something different to you (specifically because you have a clear method here).

Reply:
Ahhhh! Sorry, I'm a bit slow today. Yes, it would be good enough, as long as it could be `reset` :)